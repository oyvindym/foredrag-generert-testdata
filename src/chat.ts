import { Task } from './types';

const purge = (value: string): string => value.replace(/[-\.,]/g, ' ');

export const search = ({
  tasks,
  query,
  filter = {},
}: {
  tasks: Task[];
  query: string;
  filter?: {
    includeCompletedTasks?: boolean;
  };
}): Task[] => {
  const purgedQuery = purge(query);

  return tasks
    .filter((it) => {
      if (filter.includeCompletedTasks) {
        return true;
      }

      return !it.isCompleted;
    })
    .filter((it) => purge(it.title).indexOf(purgedQuery) !== -1);
};
