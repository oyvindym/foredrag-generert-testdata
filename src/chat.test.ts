import mockUtils from '../test/mock-utils';
import { search } from './chat';

describe('search', () => {
  test('should return task by query', () => {
    const tasks = [
      ...Array(300)
        .fill(null)
        .map(() => mockUtils.generateTask()),
      mockUtils.generateTask({
        title: 'Lag foredrag til kick-off',
        isCompleted: true,
      }),
      mockUtils.generateTask({
        title: 'Husk å ta med mac til kick off',
        isCompleted: true,
      }),
      mockUtils.generateTask({
        title: 'Husk å hold deg til å prate i 10 minutter under kick-offen.',
        isCompleted: false,
      }),
      mockUtils.generateTask({
        title: 'Få deg noe kaldt å drikke etter kick-off-foredraget',
        isCompleted: false,
      }),
    ];

    const shuffledTasks = mockUtils.shuffle(tasks);

    // expect(shuffledTasks).toMatchSnapshot();

    const searchIncompleteTasksResult = search({
      tasks: shuffledTasks,
      query: 'kick-off',
      filter: {
        includeCompletedTasks: false,
      },
    });

    expect(searchIncompleteTasksResult.length).toBe(2);
    expect(searchIncompleteTasksResult).toMatchSnapshot();
  });
});
