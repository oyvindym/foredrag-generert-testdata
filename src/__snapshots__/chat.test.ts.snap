// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search should return task by query 1`] = `
Array [
  Object {
    "description": "Itaque excepturi aut aliquid hic porro eius. Dolores ullam aut id sit adipisci voluptatem. Mollitia aut aut sed doloribus atque velit non eos.",
    "dueDate": "2026-06-27T14:03:33.823Z",
    "id": "839f6515-5497-4db6-a37d-979e1601293c",
    "isCompleted": false,
    "labels": Array [
      "aggregate",
    ],
    "priority": "LOW",
    "title": "Husk å hold deg til å prate i 10 minutter under kick-offen.",
  },
  Object {
    "description": "Ipsa sapiente veniam ut voluptates. Est est culpa dolore.",
    "dueDate": "2029-08-19T15:19:54.267Z",
    "id": "d10fea48-664a-40ff-b1b0-3cdb95005082",
    "isCompleted": false,
    "labels": Array [
      "strategize",
      "maximize",
      "integrate",
      "incentivize",
    ],
    "priority": "MEDIUM",
    "title": "Få deg noe kaldt å drikke etter kick-off-foredraget",
  },
]
`;
